### DISCUSSION TITLE
Paper cuts

### DISCUSSION THREAD

**Discussion by bmesuere** [dodona-edu/feedback-and-support#24] [General]
This discussion aims to collect smaller (UX) issues I encounter when using Dodona for the first time as a teacher. I will create a new reply for each issue. Threads can be used to discuss the individual remarks and can be converted to issue if valid.

**Comment by bmesuere** [comment]
In the dirconfig example on the docs website, both `contact` and `author` are listed. Only `contact` is listed on the page, so I think `author` should be removed here. (I would use the value from author).

https://docs.dodona.be/nl/references/exercise-config/#voorbeeld-mappenconfiguratiebestand-dirconfig-json

**Reply by BTWS2** [reply]
https://github.com/dodona-edu/dodona-edu.github.io/pull/360

**Comment by bmesuere** [comment]
In the exercise config file, the judge needs to be specified in the evaluation key under `handle`. Why is this not `judge`?
Similarly, a reading activity is still called `content` in the config file (which is an old name).

**Reply by BTWS2** [reply]
What is the new name of a reading activity?
Do you mean "`handler`"? It's also used here https://docs.dodona.be/nl/references/tested/exercise-config/#volledig-voorbeeld Can "handler" be changed to "judge"?

**Comment by bmesuere** [comment]
In the exercise config file, the user needs to list a programming language and judge. However, there is no list of valid values. Can we link to a page on Dodona containing all possible values? (e.g. the index pages of judge and programming language).

**Reply by BTWS2** [reply]
https://docs.dodona.be/en/references/exercise-config/ and https://github.com/dodona-edu/dodona-edu.github.io/pull/360

**Comment by bmesuere** [comment]
The placeholder for the clone url on the repositories/new page mentions github.ugent.be. While correct, github.com would be more logical.

**Reply by bmesuere** [reply]
Fixed in #4383

**Comment by bmesuere** [comment]
When adding a repository where Dodona doesn't have the right permissions, an error is shown:

> Clone URL ERROR: Repository not found. fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.

We could be more helpful by pointing to the exact paragraph in the documentation where it is explained which user/key needs to be added.

In addition, we should mention there (or in the docs) that it takes some time before the access request is granted (since we have to click the accept link).

**Reply by bmesuere** [reply]
Fixed in #4383

**Comment by bmesuere** [comment]
The learning activities table (at least the one on the repo page) has no empty state:
<details>
  <summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/481872/207354466-5d40cbcc-23b7-4399-95a8-c3f9e4065832.png)

</details>

**Reply by bmesuere** [reply]
Fixed in #4383

**Comment by bmesuere** [comment]
When adding a repository, the existing exercises were not added automatically. I had to trigger the webhook manually.

**Reply by bmesuere** [reply]
Fixed in #4383

**Comment by bmesuere** [comment]
Even though I followed the contact format described in the config, the rendering is a bit strange. I would have expected just my name (and the email behind the mailto link).

<details>
  <summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/481872/207356678-dad4e604-b06a-4054-823e-ae5c12ae41c3.png)

</details>

**Comment by bmesuere** [comment]
Label casing is inconsistent. I have an exercise labeled as "BDS" and "Cassandra". In the exercise table and when clicking on "edit exercise" all casing is lost:

![image](https://user-images.githubusercontent.com/481872/207368362-d8f7a51a-2f6a-4ce0-9822-f0b232dc0c44.png)

**Reply by BTWS2** [reply]
I noticed the labels are now saved as lowercase on Dodona and show with the first letter capitalized.

**Comment by bmesuere** [comment]
For an exercise without a readme, the info page provides a link that claims to be a way of easily creating such file. It however renders a 404 on github: https://github.com/bmesuere/BDS/new/master/?filename=labs/cassandra/1-installation/readme.en.md

**Reply by BTWS2** [reply]
I think it's because of "master" instead of "main".

**Comment by bmesuere** [comment]
callout-success doesn't seem to work in exercise descriptions, but is mentioned in the docs. info, warning and danger do work.

**Reply by bmesuere** [reply]
Fixed in #4385

**Comment by bmesuere** [comment]
When completing a reading activity, the message reads ` Marked as read at December 14, 2022 08:55`. That at should probably be `on`.

**Reply by bmesuere** [reply]
Fixed in #4386

**Comment by bmesuere** [comment]
It would be great if the side navigation could display the exercise state.


<details>
  <summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/481872/207543087-dc0ae353-3f6a-4e64-97f9-29ffc7d3a492.png)

</details>

**Comment by bmesuere** [comment]
On the exercise info page, the bottom table has a smaller width (on smaller screens). Do we list this table for all teachers? I would at least restrict it to "owners" of the exercise repo.

<details>
  <summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/481872/207549533-6e628261-a927-478e-b300-cc2feaa979cd.png)

</details>

**Comment by bmesuere** [comment]
On the info page of an exercise, you have the option to hand in a sample solution. This takes you to the exercise page and fills out the submission form with the solution. You then have to click submit yourself.

It would be nice if the page automatically scrolled down to the submission area instead of staying on top of the page. We already have an anchor in place we could use for this but as an added difficulty, because of the iframe, the full height of the exercise description might change after initial load.

**Comment by bmesuere** [comment]
I started using the grading module and noticed a few small issues.
- On the creation screen, it is not super clear that the prefilled deadline is the one of the series
- Reading activities are not included. It makes no sense to add comments to those, but for grading it might me important to take these into account. Should be discussed further.
- The sticky behaviour of the right grading column is broken when you scroll to the bottom of the page. It think it has to do with the footer of the site
- When you are at the last submission of an exercise, the "next" button is simply disabled. I would expect it to show a clearer message or to take me to the grading overview.
